gk/fib.c

```diff
@@ -1294,6 +1436,28 @@ add_fib_entry(const char *ip_prefix, enum gk_fib_action action,
 		"gk: add a fib entry [ip prefix = %s, action = %u, grantor_or_gateway = %s]\n",
 		ip_prefix, action, grantor_or_gateway);
 
+	/*
+	 * Take care of the LLS communication
+	 * while editing the LPM table.
+	 *
+	 * Notice, we only need to do the LLS communication
+	 * when editing the nexthop fibs.
+	 *
+	 * We use the first lcore id as the parameter
+	 * to request the resolution.
+	 */
+	if (gw_fib != NULL) {
+		ret = gk_hold_arp_and_nd(gw_fib, gk_conf->lcores[0]);
+		if (ret < 0)
+			goto out;
+	}
+
+	if (gt_fib != NULL) {
+		ret = gk_hold_arp_and_nd(gt_fib, gk_conf->lcores[0]);
+		if (ret < 0)
+			goto out;
+	}
+
```

Comment:
Why do these calls down here? Why not just call `gk_hold_arp_and_nd()` for each case inside of the `GK_FWD_GRANTOR` and `GK_FWD_GATEWAY_*_NET` cases above?